### PR TITLE
Update contributing doc

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -173,34 +96,18 @@ times in N different commits. If there was some accidental reformatting or white
...
-If an issue is complex, consider filing it and giving us time to respond before sending a
-corresponding PR. If you want to work on the issue, just let it be known on the issue thread. Giving
-us a chance to review the issue will help save time. For example, we might let you know why existing
-behavior can't be changed or about particular implementation constraints you need to keep in mind, etc.
+PR Feedback
+===========
 
-Don't feel obliged to match every issue with a PR. Simply filing issues for problems you
-encounter is a great way to contribute too! 
+Team and community members will provide feedback on your change. Community feedback is highly valued. You will often see the absense of team feedback if the community has already provided good review feedback.
```
Comment (MattGal) [inline Contributing.md]: typo (absence)

File: Contributing.md [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+2. Create a personal fork of the repository on GitHub (if you don't already have one).
+3. Create a branch off of master (`git checkout -b mybranch`). 
+  - Name the branch so that it clearly communicates your intentions, such as issue-123 or githubhandle-issue. 
+  - Branches are useful since they isolate your changes from incoming changes from upstream. They also enable you to create multiple PRs from the same fork.
+4. Make and commit your changes.
+  - Please format your commits using this [commit messages style](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+5. Add new tests corresponding to your change, if applicable.
+6. Build the repository with your changes.
+  - Make sure that the builds are clean.
+  - Make sure that the tests are are passing, including your new tests.
```
Comment (weshaggard) [inline Contributing.md]: "are all passing"

File: Contributing.md [hunk 3]
```diff
@@ -1,17 +1,39 @@
...
+  - Reuse an existing issue on the topic, if there is one.
+  - Get agreement from the team and the community that your proposed change is a good one.
+  - If your change adds a new API, follow the [[API Review Process]]. 
+  - Clearly state that you are going to take on implementing it, if that's the case. The issue filer and the implementer don't have to be the same person.
+2. Create a personal fork of the repository on GitHub (if you don't already have one).
+3. Create a branch off of master (`git checkout -b mybranch`). 
+  - Name the branch so that it clearly communicates your intentions, such as issue-123 or githubhandle-issue. 
+  - Branches are useful since they isolate your changes from incoming changes from upstream. They also enable you to create multiple PRs from the same fork.
+4. Make and commit your changes.
+  - Please format your commits using this [commit messages style](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
```
Comment (weshaggard) [inline Contributing.md]: I would reference our section below for this.

File: Contributing.md [hunk 4]
```diff
@@ -22,136 +44,37 @@
 * **DON'T** surprise us with big pull requests. Instead, file an issue and start
   a discussion so we can agree on a direction before you invest a large amount
   of time.
-* **DON'T** commit code that you didn't write. If you find MIT or Apache 2 licensed code that you think is a good fit to add to .NET Core, file an issue and start a discussion before proceeding.
+* **DON'T** commit code that you didn't write. If you find code that you think is a good fit to add to .NET Core, file an issue and start a discussion before proceeding.
+* **DON'T** submit PRs that alter licensing related files or headers. If you believe there's a problem with them, file an issue and we'll be happy to discuss it.
 * **DON'T** add API additions without filing an issue and discussing with us first.
```
Comment (weshaggard) [inline Contributing.md]: I would link to our api review process here.

**Pull request merged**